src/devices/Pcd8544/Pcd8544.cs

```diff
@@ -0,0 +1,487 @@
+                SpiWrite(false, new byte[] { (byte)(_enabled ? FunctionSet.PowerOn | FunctionSet.ExtendedMode : FunctionSet.PowerOff | FunctionSet.ExtendedMode), (byte)(0x10 | _bias) });
+            }
+        }
+
+        /// <summary>
+        /// True to inverse the screen color
+        /// </summary>
+        public bool InverseMode
+        {
+            get => _invd;
+            set
+            {
+                _invd = value;
+                SpiWrite(false, _invd ? new byte[] { (byte)(_enabled ? FunctionSet.PowerOn : FunctionSet.PowerOff), (byte)DisplayControl.InverseVideoMode } : new byte[] { (byte)FunctionSet.PowerOn, (byte)DisplayControl.NormalMode });
+            }
+        }
+
+        /// <summary>
+        /// Get or set the contrast from 0 to 127
+        /// </summary>
+        public byte Contrast
+        {
+            get => _contrast;
+            set
+            {
+                _contrast = value <= 127 ? value : throw new ArgumentException($"Contrast can only be between 0 and 127");
+                SpiWrite(false, new byte[] { (byte)(_enabled ? FunctionSet.PowerOn | FunctionSet.ExtendedMode : FunctionSet.PowerOff | FunctionSet.ExtendedMode), (byte)(0x80 | _contrast) });
+            }
+        }
+
+        /// <summary>
+        /// Get or set the temperature coefficient
+        /// </summary>
+        public Temperature Temperature
+        {
+            get => _temperature;
+            set => SpiWrite(false, new byte[] { (byte)(_enabled ? FunctionSet.PowerOn | FunctionSet.ExtendedMode : FunctionSet.PowerOff | FunctionSet.ExtendedMode), (byte)value });
+        }
+
+        #endregion
+
+        #region Primitive methods
+
+        /// <summary>
+        /// Refresh the screen
+        /// </summary>
+        public void Refresh() => SpiWrite(true, _byteMap);
+
+        /// <summary>
+        /// Clear the screen
+        /// </summary>
+        public void Clear()
+        {
+            for (int i = 0; i < _byteMap.Length; i++)
+            {
+                _byteMap[i] = 0;
+            }
+
+            SetCursorPosition(0, 0);
+            Refresh();
+        }
+
+        /// <summary>
+        /// Set the byte map
+        /// </summary>
+        /// <param name="byteMap">A 504 sized byte representing the full image</param>
+        public void SetByteMap(ReadOnlySpan<byte> byteMap)
+        {
+            if (byteMap.Length != ScreenSizeBytes)
+            {
+                throw new ArgumentException($"{nameof(byteMap)} length have to be {ScreenSizeBytes} bytes");
+            }
+
+            SetCursorPosition(0, 0);
+            byteMap.CopyTo(_byteMap);
+        }
+
+        #endregion
+
+        #region Text
+
+        /// <summary>
+        /// Write text
+        /// </summary>
+        /// <param name="text">The text to write</param>
+        public void Write(string text)
+        {
+            byte[] letter = new byte[6];
+            foreach (char c in text.ToCharArray())
```

Comment:
1. you don't need to call ToCharArray, you can enumerate chars directly
2. You can use `ReadOnlySpan<char>` rather than string. String will cast correctly

Reply:
This function is coming from the ICharacterLcd. Fine to adjust anything but it should be consistant.